UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Helpers/NMHelper.cs

```diff
@@ -181,5 +181,48 @@ public static List<ShowFilesResponseModel> ListFilesResponseCumulative(List<Batc
                 .OrderByDescending(x => Convert.ToDateTime(x.Attributes.FirstOrDefault(y => y.Key == "Data Date")?.Value))
                 .ToList();
         }
+
+        public static List<ShowFilesResponseModel> GetShowAnnualFilesResponse(List<BatchDetails> batchDetails)
+        {
+
+            List<ShowFilesResponseModel> listShowFilesResponseModels = new();
+            foreach (BatchDetails item in batchDetails)
+            {
+                foreach (BatchDetailsFiles file in item.Files)
+                {
+                    listShowFilesResponseModels.Add(new ShowFilesResponseModel
```

Comment:
Please handle duplicate batch scenario where latest batch files should be returned.

Reply:
Handled duplicate data and also added respective test cases